### PR TITLE
ci: clean up and lock benchmark node version

### PR DESCRIPTION
**Pull request by fi3ework**
## Summary

Ref: https://github.com/web-infra-dev/rslib/pull/266#issuecomment-2404134989

- lock benchmark node version
- run integration tests on 18 20 22
- clean ci

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: .github/workflows/test-ubuntu.yml [hunk 1]
```diff
@@ -76,40 +96,53 @@ jobs:
...
 
+  # ======== benchmark ========
+  benchemark-ubuntu:
+    runs-on: ubuntu-latest
+    needs: [integration-e2e-ubuntu]
+    strategy:
+      matrix:
+        # Only Ubuntu 20.04 and 22.04 are supported at the moment.
+        # See https://github.com/CodSpeedHQ/action/blob/016456b513677f9d4a1c509c7f8a38d8dd55b2b0/.github/workflows/ci.yml#L19.
+        node-version: [20.04]
```
Comment (SoonIter) [inline .github/workflows/test-ubuntu.yml:122]: `node-version: [20.04]` will pin the `ubuntu-20.04`? 😲

Comment (fi3ework) [reply .github/workflows/test-ubuntu.yml]: Yes, I misread it.

**Comment by Timeless0911** [comment]
https://github.com/web-infra-dev/rslib/settings/branch_protection_rules/50836504   we should modify this settings if we rename the ci pipeline

image

**Comment by fi3ework** [comment]
@Timeless0911 Got it, I'll modify it after PR got approved.

**Review by SoonIter — APPROVED** [review]
LGTM

**Pull request merged**